packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -0,0 +1,275 @@
+import React, { useEffect, useState, useRef, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useId } from '../helpers/useId';
+import DropdownIcon from '@strapi/icons/FilterDropdownIcon';
+import {
+  getActionFromKey,
+  getUpdatedIndex,
+  isScrollable,
+  maintainScrollVisibility,
+  MenuActions,
+  filterOptions,
+} from './utils';
+
+import { Row } from '../Row';
+import { CaretBox } from '../Select/components';
+import { Popover } from '../Popover';
+import { Box } from '../Box';
+import { Text } from '../Text';
+
+const MainRow = styled(Row)`
+  position: relative;
+  border: 1px solid ${({ theme, hasError }) => (hasError ? theme.colors.danger600 : theme.colors.neutral200)};
+  padding-right: ${({ theme }) => theme.spaces[3]};
+  padding-left: ${({ theme }) => theme.spaces[3]};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background: ${({ theme }) => theme.colors.neutral0};
+  overflow: hidden;
+
+  ${({ theme, disabled }) =>
+    disabled
+      ? `
+    color: ${theme.colors.neutral600};
+    background: ${theme.colors.neutral150};
+  `
+      : undefined}
+
+  &:focus-within {
+    border: 1px solid ${({ theme }) => theme.colors.primary600};
+  }
+`;
+
+const Input = styled.input`
+  min-height: ${40 / 16}rem;
+  border: none;
+  flex: 1;
+  font-size: ${14 / 16}rem;
+  color: ${({ theme }) => theme.colors.neutral800};
+  &:focus-visible {
+    outline: none;
+    box-shadow: none;
+    outline-offset: 0;
+  }
+`;
+
+const OptionBox = styled(Box)`
+  width: 100%;
+  border: none;
+  text-align: left;
+  outline-offset: -3px;
+  ${({ isSelected, theme }) => isSelected && `background: ${theme.colors.primary100};`}
+
+  &:hover {
+    background: ${({ theme }) => theme.colors.primary100};
+  }
+`;
+
+export const Combobox = ({ label, options, value, onChange, placeholder }) => {
```

Comment:
I think it misses the disabled props here. We don't manage disabled using plain HTML but instead rely on `aria-disabled` and prevent the onChange event from being triggered :)